main.go

```diff
@@ -83,3 +117,19 @@ func VerboseLog(msg string) {
 		fmt.Println("*", msg)
 	}
 }
+
+func WriteStringProperty(b *bytes.Buffer, name string, value string) {
+	if value != "" {
+		WriteProperty(b, name, value)
+	}
+}
+
+func WriteUInt8Property(b *bytes.Buffer, name string, value uint8) {
+	if value != 0 {
```

Comment:
Isn't 0 a valid value for some fields (say, correlation_id=0)?

Maybe there's another way to ask whether a field exists or not?